src/messages.rs

```diff
@@ -92,20 +98,18 @@ impl ErrorReturn {
 pub enum MessageType {
     ConnectRequest(ConnectRequest),
     ConnectResponse(ConnectResponse),
-    FindGroup(NameType /* Redundant, it's already in the destination */),
+    FindGroup,
     FindGroupResponse(Vec<PublicId>),
     GetData(DataRequest),
     GetDataResponse(GetDataResponse),
     DeleteData(DataRequest),
     DeleteDataResponse(ErrorReturn),
-    GetKey,
-    GetKeyResponse(NameType, sign::PublicKey),
     GetGroupKey,
-    GetGroupKeyResponse(Vec<(NameType, sign::PublicKey)>),
+    GetGroupKeyResponse(BTreeMap<NameType, sign::PublicKey>),
     Post(Data),
-    PostResponse(ErrorReturn),
+    PostResponse(ErrorReturn, BTreeMap<NameType, sign::PublicKey>),
```

Comment:
The idea is to not ask the network to pick members of the group twice. With non-response messages, any set of nodes {N1,...,Nk} can claim that they are a close group, so we can't trust that they really are. To fix that, we send the GetGroupKey message and the _network_ decides who is in the group by only sending the message to them (intermediate nodes will also catch this, but that is why we have the quorum). With response messages, we don't need to do this step, because we've already done it when we sent the request.